azerothcore/azerothcore-wotlk
# fix(DB/smart_scripts): Ulduar- Iron Mender casting Fuse Metal out of combat

## CHANGES PROPOSED:
-  Create SAI for Iron Mender to cast Fuse Metal on Iron Golem out of combat every 20 seconds
- [Proof that this event exists.](https://www.youtube.com/watch?v=m4fDOMYyjWw&ab_channel=AlValentyn)  Notice how the Golems glow blue periodically, and that the menders are channeling on them.

## TESTS PERFORMED:
- ![image](https://i.imgur.com/4FcdIv2.jpg)


## HOW TO TEST THE CHANGES:
- **Apply SQL found at #3800 first!**
- Apply this SQL
- Run Ulduar, kill the first two bosses, teleport to antechamber and watch the menders.  Pull the mobs to confirm that they don't cast every 20 second in combat, and then damage the golems to confirm that they still cast when scripted to (loss of health)

## KNOWN ISSUES AND TODO LIST:
- The timing isn't exact, in fact I did this all mostly from memory.  If anyone has any solid data on this, please feel free to modify this.  Otherwise, this serves as a base that the event does in fact exist, and we have it semi-implemented.


## Target branch(es):
